src/front/glsl/constants.rs

```diff
@@ -64,6 +66,28 @@ impl<'a> ConstantSolver<'a> {
 
                 self.access(base, self.constant_index(index)?)
             }
+            Expression::Splat {
+                size,
+                value: splat_value,
+            } => {
+                let tgt = self.solve(splat_value)?;
+                let ty = match self.constants[tgt].inner {
+                    ConstantInner::Scalar { ref value, width } => {
+                        let kind = value.scalar_kind();
+                        self.types
+                            .fetch_if(|t| t.inner == crate::TypeInner::Vector { size, kind, width })
+                    }
+                    ConstantInner::Composite { .. } => None,
+                };
+                Ok(self.constants.fetch_or_append(Constant {
+                    name: None,
+                    specialization: None,
+                    inner: ConstantInner::Composite {
+                        ty: ty.ok_or(ConstantSolvingError::SplatType)?,
```

Comment:
This is checking if the type is registered. If we just have an expression, its type is not guaranteed to be registered in the type arena. In this case, we are talking about the `Splat` expression. We don't want to panic, strictly speaking, because Naga does not need to panic at all.